pennylane/ops/op_math/prod.py

```diff
@@ -0,0 +1,281 @@
+    >>> prop_op
+    PauliX(wires=[0]) @ PauliZ(wires=[0])
+    >>> qml.matrix(prop_op)
+    array([[ 0,  -1],
+           [ 1,   0]])
+    >>> prop_op.terms()
+    ([1.0], [PauliX(wires=[0]) @ PauliZ(wires=[0])])
+
+    .. details::
+        :title: Usage Details
+
+        The Prod operator represents both matrix composition and tensor products
+        between operators.
+
+        >>> prod_op = Prod(qml.RZ(1.23, wires=0), qml.PauliX(wires=0), qml.PauliZ(wires=1))
+        >>> prod_op.matrix()
+        array([[ 0.        +0.j        ,  0.        +0.j        ,
+                 0.81677345-0.57695852j,  0.        +0.j        ],
+               [ 0.        +0.j        ,  0.        +0.j        ,
+                 0.        +0.j        , -0.81677345+0.57695852j],
+               [ 0.81677345+0.57695852j,  0.        +0.j        ,
+                 0.        +0.j        ,  0.        +0.j        ],
+               [ 0.        +0.j        , -0.81677345-0.57695852j,
+                 0.        +0.j        ,  0.        +0.j        ]])
+    """
+    _name = "Prod"
+    _eigs = {}  # cache eigen vectors and values like in qml.Hermitian
+
+    def __init__(self, *factors, do_queue=True, id=None):  # pylint: disable=super-init-not-called
+        """Initialize a Prod instance"""
+        self._id = id
+        self.queue_idx = None
+
+        if len(factors) < 2:
+            raise ValueError(f"Require at least two operators to multiply; got {len(factors)}")
+
+        self.factors = factors
+        self._wires = qml.wires.Wires.all_wires([s.wires for s in self.factors])
+
+        if do_queue:
+            self.queue()
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " @ ".join([f"{f}" for f in self.factors])
+
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.factors = tuple(f.__copy__() for f in self.factors)
+        copied_op.data = self.data.copy()  # copies the combined parameters
+
+        for attr, value in vars(self).items():
+            if attr not in {"data", "factors"}:
+                setattr(copied_op, attr, value)
+
+        return copied_op
+
+    def terms(self):  # is this method necessary for this class?
+        return [1.0], [self]
+
+    @property
+    def data(self):
+        """Create data property"""
+        return [f.parameters for f in self.factors]
+
+    @data.setter
+    def data(self, new_data):
+        """Set the data property"""
+        for new_entry, op in zip(new_data, self.factors):
+            op.data = copy(new_entry)
+
+    @property
+    def batch_size(self):
+        """Batch size of input parameters."""
+        raise ValueError("Batch size is not defined for Prod operators.")
+
+    @property
+    def ndim_params(self):
+        """ndim_params of input parameters."""
+        raise ValueError("Dimension of parameters is not currently implemented for Prod operators.")
```

Comment:
I think it should be fine to just leave these out.  Especially if we don't run `check_broadcasting` and we don't add it to the `supports_broadcasting` attribute.
```suggestion
```

Reply:
I think `batch_size` is required in order to apply `Prod` as an operation to the circuit. But I just return `None` for now since I don't think we have enough time to check if we have full support and test it before the release.

Reply:
I will remove the `ndim_params` property